bootstore/src/schemes/v0/peer.rs

```diff
@@ -686,6 +904,31 @@ impl Node {
                 }
                 self.accepted_connections.remove(&addr);
             }
+            ConnToMainMsgInner::ReceivedNetworkConfig { from, config } => {
+                let current_gen =
+                    self.network_config.as_ref().map_or(0, |c| c.generation);
+                let generation = config.generation;
+                info!(
+                    self.log,
+                    concat!(
+                        "Received network config from {} with ",
+                        "generation: {}, current generation: {}"
+                    ),
+                    from,
+                    generation,
+                    current_gen
+                );
+                if generation > current_gen {
+                    self.network_config = Some(config.clone());
+                    NetworkConfig::save(
+                        &self.log,
+                        self.config.network_config_ledger_paths.clone(),
+                        config,
+                    )
+                    .await;
+                    self.broadcast_network_config(Some(&from)).await;
+                }
```

Comment:
If someone sent us an outdated network config (`generation < current_gen`), should we send a message back to them with the current config?

Reply:
It's not necessary, as peers always send out the latest version when they learn of it or on connection. The only time this is actually likely to occur is on a connection to a peer in which case the node which has the later version has already sent it simultaneously.